docs/proposals/client-settings.md

```diff
@@ -0,0 +1,394 @@
+    // +optional
+    BodyTimeout *Duration `json:"bodyTimeout,omitempty"`
+
+    // KeepAlive defines the keep-alive settings.
+    // +optional
+    KeepAlive *KeepAlive `json:"keepAlive,omitempty"`
+}
+
+// KeepAlive defines the keepalive settings.
+type KeepAlive struct {
+    // Requests sets the maximum number of requests that can be served through one keep-alive connection.
+    // After the maximum number of requests are made, the connection is closed.
+    // +optional
+    Requests *int32 `json:"requests,omitempty"`
+
+    // Time defines the maximum time during which requests can be processed through one keep-alive connection.
+    // After this time is reached, the connection is closed following the subsequent request processing.
+    // +optional
+    Time *Duration `json:"time,omitempty"`
+
+    // Timeout defines the keep-alive timeouts.
+    // +optional
+    Timeout *KeepAliveTimeout `json:"timeout,omitempty"`
+
+    // Disable disables keep-alive connections for misbehaving browsers.
+    // +optional
+    Disable *DisableType `json:"disable"`
+}
+
+type KeepAliveTimeout struct {
+    // ServerTimeout sets the timeout during which a keep-alive client connection will stay open on the server side.
+    // The zero value disables keep-alive client connections.
+    // +optional
+    ServerTimeout *Duration
+
+    // HeaderTimeout sets the timeout in the "Keep-Alive: timeout=time" response header field.
+    // +optional
+    HeaderTimeout *Duration
+}
+
+// DisableType is the type of browsers to disable keep-alive connections on.
+type DisableType string
+
+var (
+    // NoneDisableType enables keep-alive connections for all browsers.
+    NoneDisableType DisableType = "None"
+
+    // SafariDisableType disables keep-alive connections for Safari and Safari-like browsers on macOS and macOS-like
+    // operating systems.
+    SafariDisableType DisableType = "Safari"
+
+    // MSIEDisableType disables keep-alive connections for old versions of MSIE, once a POST request is received.
+    MSIEDisableType DisableType = "MSIE"
+
+    // AllDisableType disables keep-alive connections for MSIE and Safari browsers.
+    AllDisableType DisableType = "All"
+)
+
+// Duration is a string value representing a duration in time.
+// The format is a subset of the syntax parsed by Golang time.ParseDuration.
+// Examples: 1h, 12m, 30s, 150ms.
+type Duration string
+
+// Size is a string value representing a size. Size can be specified in bytes, kilobytes (suffix k),
+// or megabytes (suffix m).
+// Examples: 1024, 8k, 1m.
+type Size string
+```
+
+### Versioning and Installation
+
+The version of the `ClientSettingsPolicy` API will be `v1alpha1`.
+
+The `ClientSettingsPolicy` CRD _may_ be installed by the Cluster Operator via Helm or with manifests. It will _not_ be required, and if the `ClientSettingsPolicy` CRD does not exist in the cluster, NGINX Gateway Fabric will not fail or log an error.
```

Comment:
I should've given this more thought. We would need to introduce a command-line flag, something like `--experimental`, in order for this to work.

Do we want to hide our alpha CRDs behind a feature flag? Or should we just require that all CRDs be installed?

Reply:
We already include steps in our docs (and our helm chart does it by default) to install the `crds/` before installing NGF, since we have the control plane CRD already.

Reply:
True, but we could add a second folder for experimental CRDs. Will users want to install alpha CRDs that they don't intend to use?

Reply:
One issue is then helm wouldn't be able to install them or treat them like CRDs if they're in a different folder, since it specifically installs from the `crds/` folder.

IMO, I don't see much issue in the CRDs being installed but not used.

Reply:
> One issue is then helm wouldn't be able to install them or treat them like CRDs if they're in a different folder, since it specifically installs from the crds/ folder.

Ah, ok. Good to know.

Reply:
Seems like there's a consensus that all our CRDs should be installed by default. I'll change this section to reflect that.